hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ClientToMemberDiscoveryTest.java

```diff
@@ -64,8 +76,14 @@ public void tearDown() {
 
     @Test
     public void clientTest() {
-        instance1 = factory.newHazelcastInstance(serverConfig);
-        instance2 = factory.newHazelcastInstance(serverConfig);
+        if (safeSerialization != null) {
+            serverConfig.getNetworkConfig().getJoin().getDiscoveryConfig().getDiscoveryStrategyConfigs().iterator().next()
+                    .addProperty("safe-serialization", safeSerialization.toString());
+            clientConfig.getNetworkConfig().getDiscoveryConfig().getDiscoveryStrategyConfigs().iterator().next()
+                    .addProperty("safe-serialization", safeSerialization.toString());
+        }
+        factory.newHazelcastInstance(serverConfig);
+        factory.newHazelcastInstance(serverConfig);
```

Comment:
What do the error messages look like when there is a configuration mismatch between clients and servers? (it is enabled on the server, but the user forgets to enable it on the client)

Reply:
Users without the `safe-serialization` will see the warning message in the log on both sides: client and member.

```
The MulticastDiscoveryStrategy Hazelcast member discovery strategy is configured without the safe-serialization parameter enabled. Set the safe-serialization property to 'true' in the strategy configuration to protect the cluster against untrusted deserialization attacks.
```

If the `safe-serialization` value doesn't match on client and member, the discovery fails:
```
09:49:55,412  INFO |clientSerializationMismatchTest[1: safeSerialization: false]| - [ClientConnectionManager] clientSerializationMismatchTest - hz.client_3 [dev] [5.1-SNAPSHOT] Trying to connect to cluster: dev
09:49:57,267  WARN |clientSerializationMismatchTest[1: safeSerialization: false]| - [ClientConnectionManager] clientSerializationMismatchTest - hz.client_3 [dev] [5.1-SNAPSHOT] Unable to get live cluster connection, retry in 1000 ms, attempt: 1, cluster connect timeout: INFINITE, max backoff: 30000 ms
...
```